test/extended/router/router.go

```diff
@@ -50,9 +50,17 @@ var _ = g.Describe("[sig-network][Feature:Router][apigroup:operator.openshift.io
 	oc = exutil.NewCLIWithPodSecurityLevel("router-stress", admissionapi.LevelBaseline)
 
 	g.BeforeEach(func() {
-		var err error
-		host, err = exutil.WaitForRouterServiceIP(oc)
+		exists, err := exutil.DoesApiResourceExist(oc.AdminConfig(), "ingresscontrollers")
```

Comment:
Does absence of `ingresscontrollers` apigroup implies existence of `router-internal-default` service?

Reply:
it does not, I guess it depends on the environment. In microshift they use this service. I suppose if it gets changed or some other env has a different name they would have to go here and customize it further

Reply:
@Miciah is it safe to fallback to the internal router service when ingresscontrollers CR is not available? I.e. is existence of ingresscontrollers cr always guaranteed when a cluster has other than internal router service?

Reply:
@dhellmann would you please help here? More elaborate on how the `router-internal-default` router service is managed? Non-existence of the `ingresscontrollers` CR is a weak condition for existence of `router-internal-default` without additional knowledge about how the service is managed. Is https://github.com/openshift/microshift/blob/main/assets/components/openshift-router/service-internal.yaml service manifest managed?

Reply:
from what I can tell the service is just created by microshift when it starts here https://github.com/openshift/microshift/blob/18491380d2e359465a47c6f23194bc1a4f1f9271/pkg/components/controllers.go#L149